drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,959 @@
+			SET_MASK(pComPort->UMCR, MCR_AFE);
+			ucTemp = UART2_MSR;             /* Read Clear */
+		}
+	#endif  /* DEF_UART2_ENABLE_CTS_RTS */
+	} else if (chan == t_uart_0) {
+		#if DEF_UART1_ENABLE_CTS_RTS
+		if (mComPortTb[chan].cts) {
+			*mComPortTb[chan].cts = 0x00;
+		}
+
+		if (mComPortTb[chan].rts) {
+			*mComPortTb[chan].rts = 0x00;
+			SET_MASK(pComPort->UMCR, MCR_AFE);
+			ucTemp = pComPort->UMSR;                /* Read Clear */
+		}
+		#endif /* DEF_UART1_ENABLE_CTS_RTS */
+	}
+
+	if (mComPortTb[chan].rx) {
+		*mComPortTb[chan].rx = 0x00;
+	}
+
+	if (mComPortTb[chan].tx) {
+		*mComPortTb[chan].tx = 0x00;
+	}
+
+	#if DEF_ENABLE_CTS_RTS
+	if (mComPortTb[chan].cts) {
+		*mComPortTb[chan].cts = 0x00;
+	}
+
+	if (mComPortTb[chan].rts) {
+		*mComPortTb[chan].rts = 0x00;
+		SET_MASK(pComPort->UMCR, MCR_AFE | MCR_RTS);
+		pComPort->URTR  = 56;
+	}
+	#endif /* DEF_ENABLE_CTS_RTS */
+
+	/* set baudrate */
+	#if CONFIG_ITE_UART_NS16550_DMA_MODE
+	pComPort->UBRCDIR = pBR->baud_int;
+	pComPort->UBRCDDR = pBR->baud_den;
+	pComPort->UBRCDNR = pBR->baud_num;
+	#endif /* CONFIG_ITE_UART_NS16550_DMA_MODE */
+
+	SET_MASK(pComPort->ULCR, LCR_DLAB);
+	pComPort->UDLL = pBR->baud_udll;
+	pComPort->UDLM = pBR->baud_udlm;
+	pComPort->UFCR = FCR_FIFO_EN | FCR_RFRST | FCR_XFRST |
+			 FCR_64B_FIFO_EN | FCR_FIFO_8;
+	pComPort->ULCR = LCR_8N1 | (bIsEnableParity ? LCR_PEN : 0) |
+			 (bIsEvenParity ? LCR_EPS : 0) |
+			 (bIsOneStopBit ? 0 : LCR_STB);
+
+	ISR4  |= BIT(6 + chan);
+	SET_MASK(pComPort->UMCR, MCR_OUT2);
+}
+
+
+static int get_br_idx(uint32_t baud_rate)
+{
+	switch (baud_rate) {
+	case 4800:
+		return t_br_4800;
+	case 9600:
+		return t_br_9600;
+	case 14400:
+		return t_br_14400;
+	case 19200:
+		return t_br_19200;
+	case 38400:
+		return t_br_38400;
+	case 57600:
+		return t_br_57600;
+	case 115200:
+		return t_br_115200;
+	case 230400:
+		return t_br_230400;
+	case 460800:
+		return t_br_460800;
+
+	default:
+		return t_br_115200;
+	}
+	;
+}
+
+#define DEV_CFG(dev)					      \
+	((const struct uart_ite_it8xxx2_device_config *const) \
+	 (dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct uart_ite_it8xxx2_dev_data_t *)(dev)->driver_data)
+
+struct uart_ite_it8xxx2_device_config {
+	uint32_t sys_clk_freq;
+};
+
+/** Device data structure */
+struct uart_ite_it8xxx2_dev_data_t {
+	uint32_t base;
+	uint32_t port;
+	uint32_t baud_rate;     /**< Baud rate */
+	uint32_t options;       /**< Serial port options */
+	uint32_t irq_no;
+	#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+	/**< cache of IIR read clear */
+	uint8_t iir_cache;
+
+	/**< Callback function pointer */
+	uart_irq_callback_user_data_t cb;
+
+	/**< Callback function arg */
+	void *cb_data;
+	#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+};
+
+static int uart_ite_it8xxx2_poll_in(struct device *dev, unsigned char *c)
+{
+	struct uart_ite_it8xxx2_dev_data_t *const dev_data = DEV_DATA(dev);
+	volatile struct T_NS16550 *pComPort =
+		(volatile struct T_NS16550 *)dev_data->base;
```

Comment:
These three lines get repeated a lot in the functions below. Have you considered declaring an inline function that takes a `struct device*` and returns a `volatile struct T_NS16550*` ?

Reply:
Excuese me, after we checking this, 
We think it couldn't be declare an inline, because the "dev_data" is also be used in the function.

Reply:
The both dev_data and the  pComPort  variable  would be used in partial dunction. (In this case, it's can't be omitted. The two lines are needed)
We remove some dummy code(pComPort) to avoid the repeats.